src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp

```diff
@@ -60,13 +66,23 @@ class Attribute {
                       });
     }
 
-    void set_callback(callback_t callback) {
-        m_callback = std::move(callback);
-    }
-
     // Apply callback to materialize RIC inside graph
-    void operator()(Input<Node> input) const {
-        m_callback(input, *this);
+    void materialize(Input<Node> input) const {
+        if (get_axis() >= input.get_partial_shape().size())
+            throw std::runtime_error("Axis not in range");
```

Comment:
it's better to use `OPENVINO_ASSERT` macro instead of `throw std::runtime_error`

Reply:
moreover it looks like we never use exceptions in transformation code, usually we just `return false`

Reply:
agree, a good sugesstion